internal/util/text.go

```
@@ -0,0 +1,184 @@
+
+// NormalizeText removes diacritics and converts special characters to ASCII-friendly equivalents for search purposes.
+func NormalizeText(text string) string {
+	if text == "" {
+		return ""
+	}
+
+	// Convert to lowercase first
+	text = strings.ToLower(text)
+
+	// Apply Unicode normalization to remove diacritics or fallback
+	diacriticRemover := transform.Chain(norm.NFD, runes.Remove(runes.In(unicode.Mn)), norm.NFC) //nolint:gochecknoglobals
```

golangci-lint (2.2.2)

directive `//nolint:gochecknoglobals` is unused for linter "gochecknoglobals"

(nolintlint)